flexmeasures/cli/data_delete.py

```diff
@@ -228,6 +228,98 @@ def delete_prognoses(
+)
+@click.option(
+    "--start",
+    "start",
+    type=AwareDateTimeField(),
+    required=False,
+    help="Remove beliefs about events starting at this datetime. Follow up with a timezone-aware datetime in ISO 6801 format.",
+)
+@click.option(
+    "--end",
+    "end",
+    type=AwareDateTimeField(),
+    required=False,
+    help="Remove beliefs about events ending at this datetime. Follow up with a timezone-aware datetime in ISO 6801 format.",
+)
+def delete_beliefs(
+    generic_assets: list[GenericAsset],
+    sensors: list[Sensor],
+    start: datetime | None = None,
+    end: datetime | None = None,
+):
+    """Delete all beliefs recorded on a given sensor (or on sensors of a given asset)."""
+
+    # Validate input
+    if not generic_assets and not sensors:
+        abort("Must pass at least one sensor or asset.")
+    elif generic_assets and sensors:
+        abort("Passing both sensors and assets at the same time is not supported.")
+    if start is not None and end is not None and start > end:
+        abort("Start should not exceed end.")
+
+    # Time window filter
+    event_filters = []
+    if start is not None:
+        event_filters += [TimedBelief.event_start >= start]
+    if end is not None:
+        event_filters += [TimedBelief.event_start + Sensor.event_resolution <= end]
+
+    # Entity filter
+    entity_filters = []
+    if sensors:
+        entity_filters += [TimedBelief.sensor_id.in_([sensor.id for sensor in sensors])]
+        select_statement = select(TimedBelief)
+    if generic_assets:
+        entity_filters += [
+            TimedBelief.sensor_id == Sensor.id,
+            Sensor.generic_asset_id.in_([asset.id for asset in generic_assets]),
+        ]
+        select_statement = select(TimedBelief, Sensor)
+
+    # Create query
+    q = select_statement.where(*entity_filters, *event_filters)
+
+    # Prompt based on count of query
+    num_beliefs_up_for_deletion = db.session.scalar(select(func.count()).select_from(q))
+    # str(asset) includes the IDs, which matters for the confirmation prompt
+    prompt = f"Delete all {num_beliefs_up_for_deletion} beliefs on sensors of {join_words_into_a_list([str(asset) for asset in generic_assets])}?"
+    click.confirm(prompt, abort=True)
+
+    # Delete all beliefs found by query
+    beliefs_up_for_deletion = db.session.scalars(q).all()
+    batch_size = 10000
+    for i, b in enumerate(beliefs_up_for_deletion, start=1):
+        if i % batch_size == 0 or i == num_beliefs_up_for_deletion:
+            click.echo(f"{i} beliefs processed ...")
+        db.session.delete(b)
+    click.secho(f"Removing {num_beliefs_up_for_deletion} beliefs ...")
+    db.session.commit()
+    num_beliefs_after = db.session.scalar(select(func.count()).select_from(q))
+    # only show the asset names for the final confirmation
+    done(
+        f"{num_beliefs_after} beliefs left on sensors of {join_words_into_a_list([asset.name for asset in generic_assets])}."
+    )
```

Comment:
In case, we want to delete beliefs only for a sensor. Then this message is not  that meaningful.

Reply:
Good catch!